test/teststl.cpp

```diff
@@ -4411,6 +4413,79 @@ class TestStl : public TestFixture {
               true);
         ASSERT_EQUALS("", errout.str());
     }
+
+    void checkMutexes() {
+        check("void f() {\n"
+              "    static std::mutex m;\n"
+              "    static std::lock_guard<std::mutex> g(m);\n"
+              "}\n",
+              true);
+        ASSERT_EQUALS("[test.cpp:3]: (warning) The lock guard won't unlock until the end of the program which could lead to a deadlock.\n", errout.str());
```

Comment:
how about "static lock guard. lock guards are intended to be local. a static lock guard could lead to a deadlock."

Reply:
Well static can be overloaded term. I would prefer "Global lock guard" to make it clear the guard is global.

Reply:
Hmm, it's still a local variable WRT `f()`.
It should have the opposite meaning of "thread local"!?